tortoise/verifying.go

```diff
@@ -32,23 +33,23 @@ func (v *verifying) resetWeights(voted types.LayerID) {
 	}
 }
 
-func (v *verifying) countBallot(logger log.Log, ballot *ballotInfo) {
+func (v *verifying) countBallot(logger *zap.Logger, ballot *ballotInfo) {
 	start := time.Now()
 
 	prev := v.layer(ballot.layer.Sub(1))
 	counted := !(ballot.conditions.badBeacon ||
 		prev.opinion != ballot.opinion() ||
 		prev.verifying.referenceHeight > ballot.reference.height)
-	logger.With().Debug("count ballot in verifying mode",
-		ballot.layer,
-		ballot.id,
-		log.ShortStringer("ballot opinion", ballot.opinion()),
-		log.ShortStringer("local opinion", prev.opinion),
-		log.Bool("bad beacon", ballot.conditions.badBeacon),
-		log.Stringer("weight", ballot.weight),
-		log.Uint64("reference height", prev.verifying.referenceHeight),
-		log.Uint64("ballot height", ballot.reference.height),
-		log.Bool("counted", counted),
+	logger.Debug("count ballot in verifying mode",
+		zap.Uint32("lid", ballot.layer.Uint32()),
+		zap.Stringer("ballot", ballot.id),
+		log.ZShortStringer("ballot opinion", ballot.opinion()),
```

Comment:
Do we need the indirection with `log.ZShortStringer`? This does the same without the need to allocate additional objects just to print a short string:

```suggestion
		zap.String("ballot opinion", ballot.opinion().ShortString()),
```

Reply:
it would even indicate that the indirection causes an additional allocation and slowdown

Reply:
that variable is not a string, but bytes - to make a short string it needs to be encoded with hex. therefore your example is not useful

Reply:
@fasmat
```go
log.Info("test", zap.String("val", s.ShortString()))
```
will call ` s.ShortString()` eagerly even if INFO is disabled.